actions/channel_actions.jsx

```diff
@@ -54,41 +53,40 @@ export async function openDirectChannelToUser(userId, success, error) {
 }
 
 export async function openGroupChannelToUsers(userIds, success, error) {
+    const state = getState();
     const result = await ChannelActions.createGroupChannel(userIds)(dispatch, getState);
     loadProfilesForSidebar();
     if (result.data && success) {
         success(result.data, false);
     } else if (result.error && error) {
-        browserHistory.push(TeamStore.getCurrentTeamUrl());
+        browserHistory.push(getCurrentTeamUrl(state));
         error({id: result.error.server_error_id, ...result.error});
     }
 }
 
 export async function loadChannelsForCurrentUser() {
-    await ChannelActions.fetchMyChannelsAndMembers(TeamStore.getCurrentId())(dispatch, getState);
+    const state = getState();
+    await ChannelActions.fetchMyChannelsAndMembers(getCurrentTeamId(state))(dispatch, getState);
     loadDMsAndGMsForUnreads();
+    loadProfilesForSidebar();
```

Comment:
Added this to fix a bug where loading a permalink directly didn't load the DMs/GMs properly until a channel switch